stubs/setuptools/setuptools/dist.pyi

```diff
@@ -1,36 +1,29 @@
-from _typeshed import Incomplete
+from collections.abc import Generator, Iterable, Mapping
+from typing import Any
 
-from setuptools import SetuptoolsDeprecationWarning
+from setuptools import Command, SetuptoolsDeprecationWarning
 
 from ._distutils.dist import Distribution as _Distribution
 
 class Distribution(_Distribution):
-    def patch_missing_pkg_info(self, attrs) -> None: ...
-    package_data: Incomplete
-    dist_files: Incomplete
-    src_root: Incomplete
-    dependency_links: Incomplete
-    setup_requires: Incomplete
-    def __init__(self, attrs: Incomplete | None = ...) -> None: ...
-    def warn_dash_deprecation(self, opt, section): ...
-    def make_option_lowercase(self, opt, section): ...
-    def parse_config_files(self, filenames: Incomplete | None = ..., ignore_option_errors: bool = ...) -> None: ...
-    def fetch_build_eggs(self, requires): ...
-    def finalize_options(self): ...
-    def get_egg_cache_dir(self): ...
+    def patch_missing_pkg_info(self, attrs: Mapping[str, Any]) -> None: ...
+    src_root: str | None
+    dependency_links: list[str]
+    setup_requires: list[str]
+    def __init__(self, attrs: dict[str, Any] | None = ...) -> None: ...
+    def warn_dash_deprecation(self, opt: str, section: str) -> str: ...
+    def make_option_lowercase(self, opt: str, section: str) -> str: ...
+    def parse_config_files(self, filenames: Iterable[str] | None = ..., ignore_option_errors: bool = ...) -> None: ...
+    def fetch_build_eggs(self, requires: str | Iterable[str]): ...
+    def get_egg_cache_dir(self) -> str: ...
     def fetch_build_egg(self, req): ...
-    def get_command_class(self, command): ...
-    def print_commands(self): ...
-    def get_command_list(self): ...
+    def get_command_class(self, command: str) -> type[Command]: ...
     def include(self, **attrs) -> None: ...
-    packages: Incomplete
-    py_modules: Incomplete
-    ext_modules: Incomplete
-    def exclude_package(self, package) -> None: ...
-    def has_contents_for(self, package): ...
+    def exclude_package(self, package: str) -> None: ...
+    def has_contents_for(self, package: str) -> bool | None: ...
     def exclude(self, **attrs) -> None: ...
-    def get_cmdline_options(self): ...
-    def iter_distribution_names(self) -> None: ...
+    def get_cmdline_options(self) -> dict[str, dict[str, str | None]]: ...
+    def iter_distribution_names(self) -> Generator[str, None, None]: ...
```

Comment:
I'd prefer to use the more general `Iterator` type here -- it's more readable, and the fact that it's specifically a generator feels like an implementation detail

```suggestion
    def iter_distribution_names(self) -> Iterator[str]: ...
```

Reply:
I'm a big fan of using `Generator`, as `Iterator` is not correct (https://github.com/python/typeshed/issues/2772), but I'm fine to stay with the current style, and this case won't affect anything.